libpromises/generic_agent.c

```diff
@@ -132,13 +134,62 @@ ENTERPRISE_VOID_FUNC_2ARG_DEFINE_STUB(void, GenericAgentSetDefaultDigest, HashMe
     *digest_len = CF_MD5_LEN;
 }
 
+void SetCFEngineRoles(EvalContext *ctx)
+{
+    char cf_hub_path[PATH_MAX];
+    StringCopy(GetBinDir(), cf_hub_path, sizeof(cf_hub_path));
+    JoinPaths(cf_hub_path, sizeof(cf_hub_path), "cf-hub");
+
+    const bool is_reporting_hub = FileCanOpen(cf_hub_path, "r");
```

Comment:
@nickanderson @vpodzime This was the "best" function I found in our file related functions. Would it be better to check if the file is executable, or if the file exists (using stat)? Do you think that's more correct? Does it make a difference in practice?

Reply:
My instinct is to use the most cheap check there is. If it does matter later, we can always improve it.

fwiw, when I implemented this in policy I did use the isexecutable() function to check.

Reply:
https://github.com/NorthernTechHQ/libntech/pull/224

Reply:
I'd argue that using `access(cf_hub_path, F_OK)` would be better here because a broken symlink would mean this is not a host collecting reports.

Reply:
@vpodzime Yea, maybe. I was thinking that it does not make sense to do more sanity checks, because once you start, where do you stop? Check if it's a file, check permissions, check if it's executable, check if it's a valid symlink, check if the binary is non-empty, run the binary with `--help` to see that it starts, check if there's also postgreSQL available and running, etc. etc.

I don't really have an opinion on whether `FileCanOpen()` or `PathExists()` or `access(cf_hub_path, F_OK)` is better, or that it makes any difference, but as Nick said maybe the fastest option is the best? `PathExists()` a.k.a. `lstat` sounds the fastest to me, but I have no data on it.